soc/xtensa/esp32/CMakeLists.txt

```diff
@@ -5,3 +5,131 @@ zephyr_sources(
+    -B ${espidf_build_dir}/bootloader -DSDKCONFIG=${espidf_build_dir}/sdkconfig
+    -DIDF_PATH=${ESP_IDF_PATH} -DIDF_TARGET=${CONFIG_BOARD}
+    -DPYTHON_DEPS_CHECKED=1
+    -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+    -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+    -DCMAKE_ASM_COMPILER=${CMAKE_ASM_COMPILER}
+    -DCMAKE_SYSTEM_NAME=${CMAKE_SYSTEM_NAME}
+    BUILD_COMMAND
+    ${CMAKE_COMMAND} --build .
+    INSTALL_COMMAND ""      # This particular build system has no install command
+    )
+
+  ExternalProject_Add(
+    EspPartitionTable
+    SOURCE_DIR ${espidf_components_dir}/partition_table
+    BINARY_DIR ${espidf_build_dir}
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND
+    python ${ESP_IDF_PATH}/components/partition_table/gen_esp32part.py -q
+    --offset 0x1000 --flash-size 4MB ${ESP_IDF_PATH}/components/partition_table/partitions_singleapp.csv ${espidf_build_dir}/partitions_singleapp.bin
+    INSTALL_COMMAND ""
+    )
+
+  if(CONFIG_BUILD_OUTPUT_BIN)
+    set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+      COMMAND python ${ESP_IDF_PATH}/components/esptool_py/esptool/esptool.py
+      ARGS --chip esp32 elf2image --flash_mode dio --flash_freq 40m
+      -o ${CMAKE_BINARY_DIR}/zephyr/${CONFIG_KERNEL_BIN_NAME}.bin
+      ${CMAKE_BINARY_DIR}/zephyr/${CONFIG_KERNEL_BIN_NAME}.elf)
+  endif()
+
+  add_dependencies(app EspIdfBootloader EspPartitionTable)
+
+  board_finalize_runner_args(esp32 "--esp-flash-bootloader=${espidf_build_dir}/bootloader/bootloader.bin")
+
+  board_finalize_runner_args(esp32 "--esp-flash-partition_table=${espidf_build_dir}/partitions_singleapp.bin")
+
+  board_finalize_runner_args(esp32 "--esp-boot-address=0x1000")
+
+  board_finalize_runner_args(esp32 "--esp-partition-table-address=0x8000")
+
+  board_finalize_runner_args(esp32 "--esp-app-address=0x10000")
+endif()
+
+if(CONFIG_BOOTLOADER_MCUBOOT)
+  include(ExternalProject)
+  include(${CMAKE_CURRENT_LIST_DIR}/dts_utils.cmake)
+  set(mcuboot_dir   ${ZEPHYR_BASE}/../bootloader/mcuboot)
+  set(mcuboot_esp_dir ${mcuboot_dir}/boot/espressif)
+  set(mcuboot_build_dir ${CMAKE_BINARY_DIR}/mcuboot)
+  set(esp32_dts ${ZEPHYR_BASE}/boards/xtensa/esp32/esp32.dts)
+
+  file(COPY ${CMAKE_CURRENT_LIST_DIR}/mcuboot.defconfig DESTINATION ${mcuboot_build_dir})
+
+  get_partition_info("${esp32_dts}" "mcuboot" mcub_off
+	  mcub_size)
+  file(APPEND ${mcuboot_build_dir}/mcuboot.defconfig
+    "CONFIG_ESP_BOOTLOADER_SIZE=${mcub_size}\n")
+
+  get_partition_info("${esp32_dts}" "image-0" img_0_off
+	  img_0_size)
+  file(APPEND ${mcuboot_build_dir}/mcuboot.defconfig
+    "CONFIG_ESP_APPLICATION_PRIMARY_START_ADDRESS=${img_0_off}\n"
+    "CONFIG_ESP_APPLICATION_SIZE=${img_0_size}\n")
+
+  get_partition_info("${esp32_dts}" "image-1" img_1_off
+	  img_1_size)
+  file(APPEND ${mcuboot_build_dir}/mcuboot.defconfig
+    "CONFIG_ESP_APPLICATION_SECONDARY_START_ADDRESS=${img_1_off}\n"
+  )
+
+  get_partition_info("${esp32_dts}" "image-scratch" img_scr_off
+	  img_scr_size)
```

Comment:
would the new dts CMake functions be helpful here ?
https://github.com/zephyrproject-rtos/zephyr/blob/164029b0c5bba375e2154678fcdd0cc0a2f76fbd/cmake/extensions.cmake#L2415-L2423

Reply:
This function is not required, existing dt APIs do the job! I have removed it from PR